WooCommerce/Classes/ViewRelated/Products/Edit Product/ProductFormViewModel.swift

```diff
@@ -432,6 +458,23 @@ extension ProductFormViewModel {
         }
     }
 
+    func saveProductImagesWhenAllAreUploaded() {
+        productImagesUploader.saveProductImagesWhenNoneIsPendingUploadAnymore(siteID: product.siteID,
+                                                                              productID: product.productID,
+                                                                              isLocalID: !product.existsRemotely) { [weak self] result in
+            guard let self = self else { return }
+            switch result {
+            case .success(let images):
+                let currentProduct = self.product
+                self.originalProduct = .init(product: self.originalProduct.product.copy(images: images))
+                self.product = .init(product: currentProduct.product)
```

Comment:
I think it will help to add a comment here that we are setting the product to retain the local changes. (As setting `originalProduct` resets the `product`)